util/file.go

```diff
@@ -253,20 +264,110 @@ func JoinTerraformModulePath(modulesFolder string, path string) string {
 	return fmt.Sprintf("%s//%s", cleanModulesFolder, cleanPath)
 }
 
-// Returns true if the given path contains the given folder name.
-//
-// Examples:
-//
-// PathContains("/foo/bar", "foo") => returns true
-// PathContains("/foo/bar", "baz") => returns false
-func PathContains(path string, folderName string) bool {
-	pathParts := strings.Split(path, string(filepath.Separator))
+// fileManifest represents a manifest with paths of all files copied by terragrunt.
+// This allows to clean those files on subsequent runs.
+// The problem is as follows: terragrunt copies the terraform source code first to "working directory" using go-getter,
+// and then copies all files from the directory it runs from to the above dir.
+// It works fine on the first run, but if we delete a file from the current terragrunt directory, we want it
+// to be cleaned in the "working directory" as well. Since we don't really know what can get copied by go-getter,
+// we have to track all the files we tough in a manifest. This way we know exactly which files we need to clean on
+// subsequent runs.
+type fileManifest struct {
+	Path         string
+	ManifestFile string
+	encoder      *gob.Encoder
+	fileHandle   *os.File
+}
 
-	for _, pathPart := range pathParts {
-		if pathPart == folderName {
-			return true
+// fileManifestEntry represents an entry in the fileManifest.
+// It uses a struct with IsDir flag so that we won't have to call Stat on every
+// file to determine if it's a directory or a file
+type fileManifestEntry struct {
+	Path  string
+	IsDir bool
+}
+
+// Clean will recursively remove all files specified in the manifest
+func (manifest *fileManifest) Clean() error {
+	return manifest.clean(manifest.Path)
+}
+
+// clean cleans the files in the manifest. If it has a directory entry, then it recursively calls clean()
+func (manifest *fileManifest) clean(path string) error {
+	manifestPath := filepath.Join(path, manifest.ManifestFile)
+
+	// if manifest file doesn't exist, just exit
+	if !FileExists(manifestPath) {
+		return nil
+	}
+	file, err := os.Open(manifestPath)
+	if err != nil {
+		// the manifest doesn't exist, so there's nothing we can safely clean
+		if err == os.ErrNotExist {
+			return nil
 		}
+		return err
 	}
+	defer file.Close()
+	decoder := gob.NewDecoder(file)
+	// decode paths one by one
+	for {
+		var manifestEntry fileManifestEntry
+		err = decoder.Decode(&manifestEntry)
+		if err != nil {
+			if err == io.EOF {
+				break
+			} else {
+				return err
+			}
+		}
+		if manifestEntry.IsDir {
+			if err := manifest.clean(manifestEntry.Path); err != nil {
+				return errors.WithStackTrace(err)
+			}
+		} else {
+			if err := os.RemoveAll(manifestEntry.Path); err != nil {
```

Comment:
If `manifestEntry` is a file path, shouldn't this use `os.Remove` instead of `os.RemoveAll`?

Reply:
Yes, fixed that